### PR TITLE
Added entity classes/migrations

### PR DESCRIPTION
**Pull request by vlad-bruma**

### CODE REVIEW

File: src/Entity/Affiliate.php [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+     * @ORM\Column(type="boolean")
+     */
+    private $active;
+
+    /**
+     * @var \DateTime
+     *
+     * @ORM\Column(type="datetime")
+     */
+    private $created_at;
```
Comment (gregurco) [inline src/Entity/Affiliate.php]: camelCase

File: src/Entity/Affiliate.php [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    /**
+     * @var Category[]|ArrayCollection
+     *
+     * @ORM\ManyToMany(targetEntity="Category", inversedBy="affiliates")
+     */
+    private $categories;
+
```
Comment (gregurco) [inline src/Entity/Affiliate.php]: don't use double new lines. Use only one

File: src/Entity/Affiliate.php [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+     */
+    public function __construct()
+    {
+        $this->categories = new ArrayCollection();
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getId()
```
Comment (gregurco) [inline src/Entity/Affiliate.php]: add type

File: src/Entity/Category.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+     * @param Job $job
+     * @return Category
+     */
+    public function removeJob(Job $job): self
+    {
+        if ($this->jobs->contains($job)) {
+            $this->jobs->removeElement($job);
+            // set the owning side to null (unless already changed)
+            if ($job->getCategory() === $this) {
+                $job->setCategory(null);
```
Comment (gregurco) [inline src/Entity/Category.php]: you specified that $category is not nullable in Job entity and set null here. You will have an error. In this case we will forbid to remove category with jobs. So, you don't need this logic here.

File: src/Entity/Job.php [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+     * @var string
+     * 
+     * @ORM\Column(type="string", length=255)
+     */
+    private $company;
+
+    /**
+     * @var string
+     * 
+     * @ORM\Column(type="string", length=255)
```
Comment (gregurco) [inline src/Entity/Job.php]: logo should be nullable

File: src/Entity/Job.php [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+     */
+    private $logo;
+
+    /**
+     * @var string
+     * 
+     * @ORM\Column(type="string", length=255)
```
Comment (gregurco) [inline src/Entity/Job.php]: should be nullable

**Review by gregurco — CHANGES_REQUESTED** [review]

**Comment by vlad-bruma** [comment]
Added adjustments

**Comment by gregurco** [comment]
@viLLoyd good job :+1:

**Pull request merged**